exercises/src/test/scala/fpinscala/state/RNGSpec.scala

```diff
@@ -0,0 +1,86 @@
+package fpinscala.state
+
+import org.specs2.mutable.Specification
+import org.specs2.ScalaCheck
+import org.scalacheck._
+
+class RNGSpec extends Specification with ScalaCheck {
+
+  "nonNegativeInt" should {
+    "only return positive numbers or zero" in {
+      prop { (seed: Int) => RNG.nonNegativeInt(RNG.Simple(seed))._1 >= 0 }
+    }
+  }
+
+  "double" should {
+    "return values between 0 and 1" in {
+      prop { (seed: Int) =>
+        val (d, _) = RNG.double(RNG.Simple(seed))
+        d must beBetween(0.0, 1.0).excludingEnd
```

Comment:
a bit off topic:
Intervals in spire
https://github.com/non/spire/blob/master/core/src/main/scala/spire/math/Interval.scala
http://plastic-idolatry.com/erik/nescala2015.pdf